lib/framework/data_context.dart

```diff
@@ -274,6 +274,10 @@ class DataContext {
       [SourceLocation? startLoc]) {
     try {
       _contextMap['getStringValue'] = Utils.optionalString;
+      // TODO: Added temporarily because the externalDataContext datas are not exposed when it's inside the nested callback methods
+      // Ex: Not working when this (widget.dataContext.addDataContext({'messageText': 'Data - EnsembleUI'});) inside the
+      // addPostFrameCallback method in screen.dart
+      _contextMap.addAll(Ensemble.externalDataContext);
```

Comment:
This is wrong. I don't understand why you need this. Also notification payload should be sent only ONCE and ONE ONLY, when the notification was delivered (or when you click on it).